More home disassembly ; Redesign TX_SYMBOL constants ; rename 'Kind' labels ; more wram addresses and text symbols ; Don't use the extras submodule ; Add support for disassembling poketcg-specific macros

xCrystal opened the pull request.

dannye commented:
You added gbz80disasm.py, but the pret submodule actually has a tcgdisasm.py which a slightly modified version that I made that correctly handles the bank1call macro and farcall macro

xCrystal commented:
True, good point. I've always used gbz80disasm.py for poketcg so I forgot that tcgdisasm.py existed with I made this.  Last time I checked I think it was behind in other aspects like reading ram symbols from the symfile though, but I could be wrong.

So when I have some time, I'll try looking at the changes you made in tcgdisasm.py and see if I can port them directly to the current gbz80disasm.py that I'm using (or the other way around). 

I'm also noticing that the files I added are lacking the more recent Python 3 compatibility commits, as I had been working on top of an old extras repo for poketcg (the poketcg/extras submodule used to point to a commit from like 2015 before I bumped it to just before the python 3 upgrade) and simply sticked to python 2.x without bothering to update. Not sure if it's relevant enough to bother about it anyway but just noticed it too.

So yeah, been kinda busy lately to be willing to spend time on this, so I'd say leave this PR open until I get into checking tcgdisasm.py and cleaning everything up sometime in 1-2 weeks hopefully.

dannye commented:
Okay

Yeah, I don't know the person that did the Python 3 compatibility commits, but they have only gotten in the way for me. I go back and forth between Python 2 and 3 but I usually use 2, and those commits have actually broken Python 2 compatibility so I have to fix the import lines locally when I want to use a certain script

yenatch commented:
Let's revert it. It doesn't even work properly with 3 from my experience.

dannye commented:
Okay, that would work
I have some free time this weekend to work on this
Free time is getting rare these days

xCrystal commented:
@dannye Since gbz80disasm.py was completely rewritten soon after you made tcgdisasm.py, it's hard to figure out which specific features you added. I have added the support for the ``farcall`` and ``bank1call`` macros and fixed a couple of other issues. Let me know if there's anything else I should look into.
